docs/cli/configuration.md

```diff
@@ -41,117 +41,225 @@ In addition to a project settings file, a project's `.gemini` directory can cont
 
 - [Custom sandbox profiles](#sandboxing) (e.g., `.gemini/sandbox-macos-custom.sb`, `.gemini/sandbox.Dockerfile`).
 
-### Available settings in `settings.json`:
+### Available settings in `settings.json`
```

Comment:
add documentation on the top of this file that we recently changed the format of this with a link to the old file format and the dates when the stable release will support the new file format and when it will auto-migrate users to the new file format.